enumerables.rb

```diff
@@ -1,41 +1,106 @@
+# frozen_string_literal: true
+# My each method for Ruby, you can use it by simply calling it .my_each
 module Enumerable
-    def my_each
-      return to_enum(:my_each) unless block_given?
-      
-      i = 0
-      array = to_a
-      while i < array.length
-        yield(array[i])
-        i+=1
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    array = to_a
+    while i < array.length
+      yield(array[i])
+      i += 1
     end
     array
   end
- 
 
-def my_each_with_index
+  def my_each_with_index
     return to_enum(:my_each_with_index) unless block_given?
-    
+
     i = 0
     array = to_a
     while i < array.length
       yield(array[i], i)
-      i+=1
+      i += 1
+    end
+    array
   end
```

Comment:
### #my_each_with_index

- [ ] When a block is given when self is a range returns the range itself after calling the given block once for each element in self